### PR TITLE
Added theory of 3 algorithms RNN, GAN (Generative Adversial Network),Computer Vision theory

### LINKED ISSUE #2026: Add theory about RNN
### Issue Summary

Add theory about RNN

### Issue Description

Add theory about RNN

### Proposed Solution (Optional)

_No response_

### Priority

High - Requires urgent attention

### Category

- [X] Enhancement
- [ ] Refactor
- [ ] Security
- [ ] Design
- [ ] Other

### Additional Context (Optional)

_No response_

### PR DESCRIPTION
**Pull request by shravya312**
### Description
Added theory of 3 algorithms RNN, GAN (Generative Adversial Network),Computer Vision theory

Fixes #2026 #2029 #2027 

### Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation update

### Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

**Comment by shravya312** [comment]
I have fixed 3 pr can u add level 2 label @ajay-dhangar

File: docs/algorithms/Computer-Vision-ML-Theory.md [hunk 1]
```diff
@@ -0,0 +1,84 @@
+---
+id: computer-vision-algorithm
+title: Computer Vision (CV) - An Overview
```
Comment (ajay-dhangar) [inline docs/algorithms/Computer-Vision-ML-Theory.md]: replace `title: Computer Vision (CV) - An Overview` with `title: "Computer Vision (CV) - An Overview"`

File: docs/algorithms/GAN-ML-Algorithm.md [hunk 2]
```diff
@@ -0,0 +1,108 @@
+---
+id: generative-adversarial-networks
+title: Generative Adversarial Networks (GANs) - An Overview
```
Comment (ajay-dhangar) [inline docs/algorithms/GAN-ML-Algorithm.md]: replace `title: Generative Adversarial Networks (GANs) - An Overview`  with `title: "Generative Adversarial Networks (GANs) - An Overview"`

File: docs/algorithms/RNN-ML-algorithm.md [hunk 3]
```diff
@@ -0,0 +1,81 @@
+---
+id: rnn-ml-algorithm
+title: Recurrent Neural Network (RNN) ML Algorithm
```
Comment (ajay-dhangar) [inline docs/algorithms/RNN-ML-algorithm.md]: replace `title: Recurrent Neural Network (RNN) ML Algorithm` with `title: "Recurrent Neural Network (RNN) ML Algorithm"`

**Review by ajay-dhangar — CHANGES_REQUESTED** [review]
update your code based on suggestions

**Comment by shravya312** [comment]
Okay I will make all the changes

**Comment by shravya312** [comment]
I have made the changes.Can u review the changes @ajay-dhangar

**Review by ajay-dhangar — APPROVED** [review]
Done

**Pull request merged**